src/web/lib/contributions.test.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    getLastOneOffContributionDate,
+    ONE_OFF_CONTRIBUTION_DATE_COOKIE,
+    SUPPORT_ONE_OFF_CONTRIBUTION_COOKIE,
+} from './contributions';
+
+const setCookie = (name: string, value: string) => {
+    Object.defineProperty(window.document, 'cookie', {
+        writable: true,
+        value: `${name}=${value}`,
+    });
+};
+
+const clearAllCookies = () => {
+    const cookies = document.cookie.split(';');
+
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < cookies.length; i++) {
+        const cookie = cookies[i];
+        const eqPos = cookie.indexOf('=');
+        const name = eqPos > -1 ? cookie.substr(0, eqPos) : cookie;
+        document.cookie = `${name}=;expires=Thu, 01 Jan 1970 00:00:00 GMT`;
+    }
+};
+
+describe('getLastOneOffContributionDate', () => {
+    beforeEach(clearAllCookies);
+
+    it('returns date from attributes cookie if only cookie found', () => {
+        const somePastDate = '2020-01-28';
+        const formattedPastDate = Date.parse(somePastDate).toString();
+        setCookie(ONE_OFF_CONTRIBUTION_DATE_COOKIE, somePastDate);
+        const lastOneOffContributionDate = getLastOneOffContributionDate();
+        expect(lastOneOffContributionDate).toBe(formattedPastDate);
+    });
+
+    it('returns a support cookie date if only cookie found', () => {
+        const somePastDate = '1582567969093';
```

Comment:
Is it worth scrapping this test as it only seems to be testing cookie retrieval if I've understood correctly?

Reply:
Yeah it's testing that our function will read the cookie if it's set. Not huge value but *some* value, if we were to rename the cookie or change the logic to stop reading it, or parsing it differently, the test would flag that.